### PR TITLE
cmd/mev-boost: add urfave as cli framework

### PR DESCRIPTION
**Pull request by MariusVanDerWijden**
## 📝 Summary

This PR adds urfave/cli as a cli framework and reworks some of the logic in the main package.
It also gets rid of the weird redirections into the cli package

## ⛱ Motivation and Context

Cleanup, adds a cli framework which makes adding flags easier in the future

### CODE REVIEW

**Comment by MariusVanDerWijden** [comment]
closes https://github.com/flashbots/mev-boost/issues/640

**Comment by metachris** [comment]
Cool, thanks! A few quick notes
- The "weird indirection" was to allow `go install github.com/flashbots/mev-boost` (it won't work without a root main.go file i think)
- I like urfave a lot, and the cli flags have grown significantly and unpleasantly in mev-boost. But is the switch to urfave backwards-compatible, or would it break? This seems to be a major concern, if the upgrade would force all proposers to update their cli flags.

File: Makefile [hunk 1]
```diff
@@ -24,7 +24,7 @@ v:
 .PHONY: build
 build:
 	@go version
-	CGO_ENABLED=0 go build $(GO_BUILD_FLAGS) -o mev-boost
+	CGO_ENABLED=0 go build $(GO_BUILD_FLAGS) -o mev-boost ./cmd/mev-boost
```
Comment (jtraglia) [inline Makefile:27]: What's the rationale for this change?

**Review by jtraglia — DISMISSED** [review]

**Comment by metachris** [comment]
@MariusVanDerWijden can you please add the root `main.go` file back, so `go install github.com/flashbots/mev-boost` works?

File: Makefile (hunk 1 above)
Comment (metachris) [reply Makefile:27]: removing the root `main.go` file (i think we should keep that file and not change the build)

**Review by metachris — APPROVED** [review]
